src/main/java/org/nibor/autolink/internal/HashTagScanner.java

```diff
@@ -0,0 +1,68 @@
+package org.nibor.autolink.internal;
+
+import org.nibor.autolink.LinkSpan;
+import org.nibor.autolink.LinkType;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Scan for HashTags starting from the trigger character "#"
+ * <p>
+ */
+public class HashTagScanner implements Scanner {
+
+    public static final char HASHTAG_DELIMITER = ' ';
+    private Set<Character> allowedSpecialChars;
+
+    public HashTagScanner(Set<Character> allowedSpecialChars) {
+        if (allowedSpecialChars == null) {
+            allowedSpecialChars = getDefaultAllowedSpecialChars();
+        }
+        if (allowedSpecialChars.contains(HASHTAG_DELIMITER)) {
+            throw new IllegalArgumentException("Allowed special characters cannot contain " + HASHTAG_DELIMITER);
+        }
+        this.allowedSpecialChars = allowedSpecialChars;
+    }
+
+    @Override
+    public LinkSpan scan(CharSequence input, int triggerIndex, int rewindIndex) {
+        int last = findLast(input, triggerIndex);
```

Comment:
I think we need to check preceding characters as well. E.g. Twitter doesn't seem to treat `#bar` in `foo#bar` as a hashtag.